dissect/target/plugins/os/unix/bsd/citrix/_os.py

```diff
@@ -117,10 +138,19 @@ def users(self) -> Iterator[UnixUserRecord]:
                 # the 'root' user as having '/root' as a home, not in /var/nstmp.
                 user_home = "/root"
 
+            seen.add(username)
             yield UnixUserRecord(name=username, home=user_home)
 
         for username in nstmp_users:
-            yield UnixUserRecord(name=username, home=nstmp_user_path.format(username=username))
+            seen.add(username)
+            yield UnixUserRecord(name=username, home=nstmp_path.joinpath(username))
+
+        # Yield users from /etc/shadow if we have not seem them in previous loops
+        for user in super().users():
+            if user.name in seen:
```

Comment:
You don't add these to the seen.

Reply:
As this is the final loop of the method, and we are only receiving users from the `super().users()` function, I don't think we have to add the users to the seen.